### ISSUE TITLE
`TypeError: 'str' object is not callable` when trying to add reference

### ISSUE THREAD

**Issue by beuerle** [pubs/pubs#186]
Since recently I am not able to add references to pubs anymore. I get the following error message:

```bash
$ pubs --version
0.8.2
$ pubs add -D 10.1109/TTHZ.2018.2884612
error: 'str' object is not callable
Traceback (most recent call last):
  File "/bin/pubs", line 11, in <module>
    load_entry_point('pubs==0.8.2', 'console_scripts', 'pubs')()
  File "/usr/lib/python3.7/site-packages/pubs/pubs_cmd.py", line 103, in execute
    if not uis.get_ui().handle_exception(e):
  File "/usr/lib/python3.7/site-packages/pubs/pubs_cmd.py", line 100, in execute
    args.func(conf, args)
  File "/usr/lib/python3.7/site-packages/pubs/commands/add_cmd.py", line 111, in command
    bibentry = bibentry_from_api(args, ui)
  File "/usr/lib/python3.7/site-packages/pubs/commands/add_cmd.py", line 78, in bibentry_from_api
    return apis.get_bibentry_from_api(args.doi, 'doi', ui=ui, raw=raw)
  File "/usr/lib/python3.7/site-packages/pubs/apis.py", line 56, in get_bibentry_from_api
    bibentry = endecoder.EnDecoder().decode_bibdata(bibentry_raw)
  File "/usr/lib/python3.7/site-packages/pubs/endecoder.py", line 133, in decode_bibdata
    homogenize_fields=True).get_entry_dict()
  File "/usr/lib/python3.7/site-packages/bibtexparser/bparser.py", line 78, in __new__
    return parse(data, **args)
  File "/usr/lib/python3.7/site-packages/bibtexparser/bparser.py", line 31, in parse
    return parser.parse(data)
  File "/usr/lib/python3.7/site-packages/bibtexparser/bparser.py", line 155, in parse
    self._expr.parseFile(bibtex_file_obj)
  File "/usr/lib/python3.7/site-packages/bibtexparser/bibtexexpression.py", line 281, in parseFile
    return self.main_expression.parseFile(file_obj, parseAll=True)
  File "/usr/lib/python3.7/site-packages/pyparsing.py", line 2362, in parseFile
    return self.parseString(file_contents, parseAll)
  File "/usr/lib/python3.7/site-packages/pyparsing.py", line 1804, in parseString
    loc, tokens = self._parse( instring, 0 )
  File "/usr/lib/python3.7/site-packages/pyparsing.py", line 1548, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "/usr/lib/python3.7/site-packages/pyparsing.py", line 4361, in parseImpl
    return super(ZeroOrMore, self).parseImpl(instring, loc, doActions)
  File "/usr/lib/python3.7/site-packages/pyparsing.py", line 4291, in parseImpl
    loc, tokens = self_expr_parse( instring, loc, doActions, callPreParse=False )
  File "/usr/lib/python3.7/site-packages/pyparsing.py", line 1548, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "/usr/lib/python3.7/site-packages/pyparsing.py", line 3871, in parseImpl
    ret = e._parse( instring, loc, doActions )
  File "/usr/lib/python3.7/site-packages/pyparsing.py", line 1548, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "/usr/lib/python3.7/site-packages/pyparsing.py", line 3722, in parseImpl
    loc, exprtokens = e._parse( instring, loc, doActions )
  File "/usr/lib/python3.7/site-packages/pyparsing.py", line 1548, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "/usr/lib/python3.7/site-packages/pyparsing.py", line 3871, in parseImpl
    ret = e._parse( instring, loc, doActions )
  File "/usr/lib/python3.7/site-packages/pyparsing.py", line 1548, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "/usr/lib/python3.7/site-packages/pyparsing.py", line 3722, in parseImpl
    loc, exprtokens = e._parse( instring, loc, doActions )
  File "/usr/lib/python3.7/site-packages/pyparsing.py", line 1548, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "/usr/lib/python3.7/site-packages/pyparsing.py", line 3705, in parseImpl
    loc, resultlist = self.exprs[0]._parse( instring, loc, doActions, callPreParse=False )
  File "/usr/lib/python3.7/site-packages/pyparsing.py", line 1581, in _parseNoCache
    tokens = fn( instring, tokensStart, retTokens )
  File "/usr/lib/python3.7/site-packages/pyparsing.py", line 1203, in wrapper
    ret = func(*args[limit[0]:])
  File "/usr/lib/python3.7/site-packages/bibtexparser/bibtexexpression.py", line 67, in field_to_pair
    strip_after_new_lines(f.get('Value')))
  File "/usr/lib/python3.7/site-packages/bibtexparser/bibtexexpression.py", line 25, in strip_after_new_lines
    return _strip_after_new_lines(s)
  File "/usr/lib/python3.7/site-packages/bibtexparser/bibtexexpression.py", line 10, in _strip_after_new_lines
    lines = s.splitlines()
TypeError: 'str' object is not callable
```

Required packages versions:

```
PyYAML==3.13
bibtexparser==1.0.1
python-dateutil==2.7.5
requests==2.21.0
configobj==5.0.6
beautifulsoup4==4.6.3
feedparser==5.2.1
six==1.12.0
```
To exclude that some installed packages on my system cause the problem, I also tried to install `pubs` in a virtual environment and got the same result:

```bash
$ pubs add -D 10.1109/TTHZ.2018.2884612
error: 'str' object is not callable
Traceback (most recent call last):
  File "/home/user/workspace/env/pubs/bin/pubs", line 11, in <module>
    sys.exit(execute())
  File "/home/user/workspace/env/pubs/lib/python3.7/site-packages/pubs/pubs_cmd.py", line 103, in execute
    if not uis.get_ui().handle_exception(e):
  File "/home/user/workspace/env/pubs/lib/python3.7/site-packages/pubs/pubs_cmd.py", line 100, in execute
    args.func(conf, args)
  File "/home/user/workspace/env/pubs/lib/python3.7/site-packages/pubs/commands/add_cmd.py", line 111, in command
    bibentry = bibentry_from_api(args, ui)
  File "/home/user/workspace/env/pubs/lib/python3.7/site-packages/pubs/commands/add_cmd.py", line 78, in bibentry_from_api
    return apis.get_bibentry_from_api(args.doi, 'doi', ui=ui, raw=raw)
  File "/home/user/workspace/env/pubs/lib/python3.7/site-packages/pubs/apis.py", line 56, in get_bibentry_from_api
    bibentry = endecoder.EnDecoder().decode_bibdata(bibentry_raw)
  File "/home/user/workspace/env/pubs/lib/python3.7/site-packages/pubs/endecoder.py", line 133, in decode_bibdata
    homogenize_fields=True).get_entry_dict()
  File "/home/user/workspace/env/pubs/lib/python3.7/site-packages/bibtexparser/bparser.py", line 75, in __new__
    return parse(data, **args)
  File "/home/user/workspace/env/pubs/lib/python3.7/site-packages/bibtexparser/bparser.py", line 31, in parse
    return parser.parse(data)
  File "/home/user/workspace/env/pubs/lib/python3.7/site-packages/bibtexparser/bparser.py", line 147, in parse
    self._expr.parseFile(bibtex_file_obj)
  File "/home/user/workspace/env/pubs/lib/python3.7/site-packages/bibtexparser/bibtexexpression.py", line 263, in parseFile
    return self.main_expression.parseFile(file_obj, parseAll=True)
  File "/home/user/workspace/env/pubs/lib/python3.7/site-packages/pyparsing.py", line 2362, in parseFile
    return self.parseString(file_contents, parseAll)
  File "/home/user/workspace/env/pubs/lib/python3.7/site-packages/pyparsing.py", line 1804, in parseString
    loc, tokens = self._parse( instring, 0 )
  File "/home/user/workspace/env/pubs/lib/python3.7/site-packages/pyparsing.py", line 1548, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "/home/user/workspace/env/pubs/lib/python3.7/site-packages/pyparsing.py", line 4361, in parseImpl
    return super(ZeroOrMore, self).parseImpl(instring, loc, doActions)
  File "/home/user/workspace/env/pubs/lib/python3.7/site-packages/pyparsing.py", line 4291, in parseImpl
    loc, tokens = self_expr_parse( instring, loc, doActions, callPreParse=False )
  File "/home/user/workspace/env/pubs/lib/python3.7/site-packages/pyparsing.py", line 1548, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "/home/user/workspace/env/pubs/lib/python3.7/site-packages/pyparsing.py", line 3871, in parseImpl
    ret = e._parse( instring, loc, doActions )
  File "/home/user/workspace/env/pubs/lib/python3.7/site-packages/pyparsing.py", line 1548, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "/home/user/workspace/env/pubs/lib/python3.7/site-packages/pyparsing.py", line 3722, in parseImpl
    loc, exprtokens = e._parse( instring, loc, doActions )
  File "/home/user/workspace/env/pubs/lib/python3.7/site-packages/pyparsing.py", line 1548, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "/home/user/workspace/env/pubs/lib/python3.7/site-packages/pyparsing.py", line 3871, in parseImpl
    ret = e._parse( instring, loc, doActions )
  File "/home/user/workspace/env/pubs/lib/python3.7/site-packages/pyparsing.py", line 1548, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "/home/user/workspace/env/pubs/lib/python3.7/site-packages/pyparsing.py", line 3722, in parseImpl
    loc, exprtokens = e._parse( instring, loc, doActions )
  File "/home/user/workspace/env/pubs/lib/python3.7/site-packages/pyparsing.py", line 1548, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "/home/user/workspace/env/pubs/lib/python3.7/site-packages/pyparsing.py", line 3705, in parseImpl
    loc, resultlist = self.exprs[0]._parse( instring, loc, doActions, callPreParse=False )
  File "/home/user/workspace/env/pubs/lib/python3.7/site-packages/pyparsing.py", line 1581, in _parseNoCache
    tokens = fn( instring, tokensStart, retTokens )
  File "/home/user/workspace/env/pubs/lib/python3.7/site-packages/pyparsing.py", line 1203, in wrapper
    ret = func(*args[limit[0]:])
  File "/home/user/workspace/env/pubs/lib/python3.7/site-packages/bibtexparser/bibtexexpression.py", line 68, in field_to_pair
    strip_after_new_lines(f.get('Value')))
  File "/home/user/workspace/env/pubs/lib/python3.7/site-packages/bibtexparser/bibtexexpression.py", line 25, in strip_after_new_lines
    return _strip_after_new_lines(s)
  File "/home/user/workspace/env/pubs/lib/python3.7/site-packages/bibtexparser/bibtexexpression.py", line 10, in _strip_after_new_lines
    lines = s.splitlines()
TypeError: 'str' object is not callable
```

Any ideas where the problem comes from?

**Comment by benureau** [comment]
Yes, this is due to a incompatibility in the dependencies (see https://github.com/sciunto-org/python-bibtexparser/issues/225). You could try to downgrade the PyParsing version.

**Comment by omangin** [comment]
Closing now that the upstream fix is merged ([sciunto-org/python-bibtexparser/#226](https://github.com/sciunto-org/python-bibtexparser/pull/226)). Please re-open if the bug still occurs with the latest master of bibtexparser (this means that until a new version of bibtexparser is release you have to install it through `pip install --upgrade git+https://github.com/sciunto-org/python-bibtexparser`).

**Comment by m040601** [comment]
"error: 'str' object is not callable"

I'm still getting this error message running pubs add with  -D or -I flags.

Im on ArchLinux running:
pubs --version
0.8.2

which is pulling  python-bibtexparser
Version:        1.1.0-2

**Comment by benureau** [comment]
@m040601: that's not good. When you say you are still getting the error, did you have the error before, it went away, and came back? When did `pubs add -D` worked for you the last time? Which version of python are you using exactly? Could you try installing the master of pubs and see if that fixes things? We need to release a new version, but that may take a few days.

**Comment by m040601** [comment]
> When did pubs add -D worked for you the last time

sorry, can't remember exactly. Last time I had tried pubs was at least 3 months ago. Didn't get this error (nor remember any other) then.

> Which version of python are you using exactly? 

Python 3.7.4. Always use my distros official version.

>  Could you try installing the master of pubs and see if that fixes things? 

Actually I did that. I tried both, the "stable" version,
 https://aur.archlinux.org/packages/pubs
that pulls from a tagged release.
As well as  ,
https://aur.archlinux.org/packages/pubs-git/
that pull from the latest thing you have on master.
Both give me the same error.

**Comment by m040601** [comment]
Right. Just tried something else for testing. A disposable python virtual environment.
And this time pubs gave me no error.

$ python -m venv pubsven
installs this:
$ tree -L 1 pubsvenv/
pubsvenv/
├── bin
├── include
├── lib
├── lib64 -> lib
└── pyvenv.cfg
$ source pubsvenv/bin/activate
now I can play with python without messing my system
(pubsvenv) 
pip install papis

works has expected

**Comment by m040601** [comment]
I think I'm getting closer to the problem root cause. Sorry if I brought some confusion here,  but maybe it's just my fault not pubs. Anyway, this may be usefull as a troubleshooting help for other users.

I started slowly one by one comparing the installed dependencies and its version numbers on the python virtual env.

Spotted a bibtexparser 1.1.0

Then went back to search my 
$HOME/.local/lib/python3.7/site-packages/
That's where the stuff ends up  when you do,
pip install --local somepythonapp.

And indeed found a forgottten bibtexparser 1.0.x there. Deleted and everything seems to  work fine now.

LESSON LEARNED 1: Didnt know that the python 'libs'  I have installed at a "user level" in
$HOME/.local/lib/python3.7/ take precedence over the "system" ones
/usr/lib/python3.7/site-packages/
Am I saying this right ?


LESSON LEARNED 2: Try to never ever use "pip install --local XYZ" when it's gonna pull something (important and commonly used) dependecy, that my system package manager might be able to track and manage much better.
Prefer to use a python virtual env and do a "pip install XYZ" there inside.
